OpenRA.Game/Server/Server.cs

```diff
@@ -124,6 +128,40 @@ public void EndGame()
 		{
 			foreach (var t in serverTraits.WithInterface<IEndGame>())
 				t.GameEnded(this);
+
+			if (recorder != null)
+				recorder.Dispose();
+		}
+
+		// Craft a fake handshake request/response because that's the
+		// only way to expose the Version and OrdersProtocol.
+		public void RecordFakeHandshake()
+		{
+			var request = new HandshakeRequest
+			{
+				Mod = ModData.Manifest.Id,
+				Version = ModData.Manifest.Metadata.Version,
+			};
+			recorder.ReceiveFrame(0, 0, new Order("HandshakeRequest", null, false)
```

Comment:
Style nit: Please add an empty line before this line (block-like syntax should generally have an empty line afterwards).